components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/handlers/IndexHtmlForwardHandler.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.view.handlers;
+
+import javax.annotation.Nonnull;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.Handler;
+import org.sonatype.nexus.repository.view.Request;
+import org.sonatype.nexus.repository.view.Response;
+import org.sonatype.nexus.repository.view.ViewFacet;
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+/**
+ * Handler which will forward current request to {@code <request.path>/index.html}.
+ *
+ * @since 3.0
+ */
+@Named
+@Singleton
+public class IndexHtmlForwardHandler
+    extends ComponentSupport
+    implements Handler
+{
+  @Nonnull
+  @Override
+  public Response handle(@Nonnull final Context context) throws Exception {
+    Request original = context.getRequest();
+
+    String path = original.getPath();
+
+    // sanity, to ensure we don't corrupt the path
+    if (!path.endsWith("/")) {
+      path = path + "/";
+    }
+    path = path + "index.html";
```

Comment:
Were we going to make this configurable? My sense was that the purpose of this feature was to help a raw repo and one's content play nicely. (Though, maybe the only content we ever really need to adapt to is mvn site's "index.html".)